spec/requests/providers/means_summaries_spec.rb

```diff
@@ -82,84 +84,123 @@
+          request
           expect(legal_aid_application.reload.provider_entering_merits?).to be true
         end
 
-        context "when provider does not have bank_statement_upload permissions" do
+        context "when provider has bank_statement_upload permissions" do
+          before do
+            allow(Setting).to receive(:enhanced_bank_upload?).and_return(false)
+            legal_aid_application.provider.permissions << Permission.find_or_create_by(role: "application.non_passported.bank_statement_upload.*")
+            legal_aid_application.update!(provider_received_citizen_consent: false)
+          end
+
+          it "redirects to the no eligibility assessment page" do
+            request
+            expect(response).to redirect_to(providers_legal_aid_application_no_eligibility_assessment_path(legal_aid_application))
+          end
+
+          it "does not calls CFE::SubmissionManager" do
+            request
+            expect(CFE::SubmissionManager).not_to have_received(:call)
+          end
+        end
+
+        context "when provider has bank_statement_upload permissions and enhanced_bank_upload is enabled" do
+          before do
+            allow(Setting).to receive(:enhanced_bank_upload?).and_return(true)
+            legal_aid_application.provider.permissions << Permission.find_or_create_by(role: "application.non_passported.bank_statement_upload.*")
+            legal_aid_application.update!(provider_received_citizen_consent: false)
+          end
+
+          it "redirects to the capital income assessment page" do
+            request
+            expect(response).to redirect_to(providers_legal_aid_application_capital_income_assessment_result_path(legal_aid_application))
+          end
+
+          it "calls CFE::SubmissionManager" do
+            request
+            expect(CFE::SubmissionManager).to have_received(:call).with(legal_aid_application.id)
+          end
+        end
+
+        context "when provider is not uploading bank statements" do
           before do
-            legal_aid_application.provider.permissions.find_by(role: "application.non_passported.bank_statement_upload.*")&.destroy
+            legal_aid_application.update!(provider_received_citizen_consent: true)
           end
 
           it "redirects to the capital income assessment page" do
+            request
             expect(response).to redirect_to(providers_legal_aid_application_capital_income_assessment_result_path(legal_aid_application))
           end
+
+          it "calls CFE::SubmissionManager" do
+            request
+            expect(CFE::SubmissionManager).to have_received(:call).with(legal_aid_application.id)
+          end
         end
 
-        context "Form submitted using Save as draft button" do
-          let(:params) { { draft_button: "Save as draft" } }
+        context "when form submitted using Save as draft button" do
+          let(:params) { { draft_button: "irrelevant" } }
```

Comment:
Is this because it doesn't matter what the text is on the 'Save as draft' button?

Reply:
yeah, this is [because only presence](https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/2fde11d007f1ff351d165fa84cd69b1a8dfb6e6f/app/controllers/concerns/providers/draftable.rb#L47-L49) is checked for